### PR TITLE
Use C++11 range-based for loops

### PR DESCRIPTION
**Pull request by turol**
Used clang-modernizer to refactor some for loops and then cleaned up by hand. These are only those which the modernizer considers "safe".

### CODE REVIEW

**Comment by acgessler** [comment]
@kimkulling 

I may lack some recent status here, but I thought we explicitly don't do C++11 yet.

I occasionally use MSVC11 to compile, which does not support all of it (it would support range based for loops however).

**Comment by turol** [comment]
C++11 has been required since #850 was merged. Range-based for loops are already used. I'm not sure what MSVC minimum requirement has officially been set, if any, and what C++11 features are missing. Is "auto" supported? It would allow removal of lots of iterator boilerplate.

File: code/LWOMaterial.cpp [hunk 1]
```diff
@@ -343,24 +343,24 @@ void LWOImporter::ConvertMaterial(const LWO::Surface& surf,aiMaterial* pcMat)
 
     // Now we need to know which shader to use .. iterate through the shader list of
     // the surface and  search for a name which we know ...
-    for (ShaderList::const_iterator it = surf.mShaders.begin(), end = surf.mShaders.end();it != end;++it)   {
-        //if (!(*it).enabled)continue;
```
Comment (kimkulling) [inline code/LWOMaterial.cpp]: Remove dead code.

**Comment by kimkulling** [comment]
This quetsion comes up some months ago: use c++11 or not. Most of the votes were for using it. So ...

File: code/LWOMaterial.cpp (hunk 1 above)
Comment (turol) [inline code/LWOMaterial.cpp]: It was there already. Do you know who commented it out and why?

**Comment by kimkulling** [comment]
Merged, thanks a lot!

**Pull request merged**